src/checker.c

```diff
@@ -474,6 +474,10 @@ Conversion conversion(Type *type, Type *target) {
         return result;
     }
 
+    if (type->kind == TypeKind_Enum && IsInteger(target)) {
```

Comment:
Let's use `isEnum(type)` incase we want to change the layout of `Type` in the future